manim/mobject/graphing/coordinate_systems.py

```diff
@@ -588,6 +590,8 @@ def plot(
         function: Callable[[float], float],
         x_range: Sequence[float] | None = None,
         use_vectorized: bool = False,
+        colorscale: Union[Iterable[Color], Color] | None = None,
```

Comment:
Passing a single color, which appears to be allowed based on this type hint, yields
```
ValueError: invalid literal for int() with base 16: ''
```

Also, the second option (passing an interable containing tuples of the form `(color, pivot)` is not captured by this type hint, this should be fixed.

Reply:
Is this the right way to do it?

```py
colorscale: Union[Iterable[Color], Iterable[Color, float]] | None = None,
```

Reply:
https://github.com/ManimCommunity/manim/discussions/2822